src/dataflow/src/sink/kafka.rs

```diff
@@ -694,15 +697,14 @@ fn avro_encode_stream<G>(
     input_stream: &Stream<G, ((Option<Row>, Option<Row>), Timestamp, Diff)>,
     as_of: SinkAsOf,
     gate_ts: Option<Timestamp>,
-    encoder: Encoder,
-    key_schema_id: Option<i32>,
-    value_schema_id: i32,
+    encoder: impl Encode + 'static,
     fuel: usize,
     name_prefix: String,
 ) -> Stream<G, ((Option<Vec<u8>>, Option<Vec<u8>>), Timestamp, Diff)>
 where
     G: Scope<Timestamp = Timestamp>,
 {
+    // todo change name
```

Comment:
I'm guessing you want to put something like `get_format_name()` in the `Encode` trait? I'm still not experienced enough to have opinions about good trait design.